### ISSUE TITLE
[oc_filecache] Loading one million files

### ISSUE THREAD

**Issue by dariodsa** [nextcloud/server#25995]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

As you already know, Nextcloud has issues with loading a huge number of files. As I saw, reason is that for each file, Nextcloud does ``SELECT`` query on table ``oc_filecache``. My first question i, what could be stored in ``oc_filecache`` that Nextcloud couldn't see from the disk? Second question, I would like to refactor a code so  that if ``file_cache`` is set to 0 in ``config.php``, Nextcloud wouldn't use ``oc_filecache`` at all. So I would like your opinion on that matter? That would probably load page slower, but in some case faster when number of files is huge. 

Thanks,
Dario Sindicic

**Comment by dariodsa** [comment]
#23930

**Comment by SimplyCorbett** [comment]
Nextcloud is the metadata server for your files. All files use oc_filecache because it stores the file path among other details in that table.

I personally have not experienced issues with having a huge number of files. I've had millions of files on nextcloud with no issue.

The issue you linked to #23930 concerns loading a lot of files in one folder within the WebGUI... this is not the same as nextcloud being unable to handle millions of files.

Further, if you are using webdav nextcloud you can easily load folders with millions of files. I've tested this on linux with dolphin. The problem is your (local) computer is going to have RAM and CPU usage through the roof trying to browse that folder.

tl;dr explain your problem in more detail.

**Comment by dariodsa** [comment]
I analyse more detailed my environment and I saw that I am having performance issue with file locking using Redis. When file locking is disabled, Nextcloud renders page really quickly, but when file locking is enabled it simply times out(on folder with more than 7000 files). Is there something special which I should put in my Redis configuration?

**Comment by szaimen** [comment]
Lets track this in https://github.com/nextcloud/server/issues/7312
